petdeface/petdeface.py

```diff
@@ -212,85 +220,82 @@ def init_single_subject_wf(subject_id: str, bids_dir: str) -> Workflow:
         workflow for subject
     """
     name = f"single_subject_{subject_id}_wf"
-    subject_data = collect_data(
-        bids_dir,
-        subject_id,
-    )[0]
 
-    if not subject_data["pet"]:
-        raise RuntimeError(
-            "No PET images found for participant {}. "
-            "All workflows require PET images.".format(subject_id)
-        )
+    if isinstance(bids_data, pathlib.Path):
+        bids_data = BIDSLayout(bids_data)
+    elif isinstance(bids_data, BIDSLayout):
+        pass
 
-    if not subject_data["t1w"]:
-        raise RuntimeError(
-            "No T1w images found for participant {}. "
-            "All workflows require T1w images.".format(subject_id)
-        )
+    data = collect_anat_and_pet(bids_data)
```

Comment:
Looks like the pet-mri matching is now being done by this petutils function. Cool!